modules/command/dc/dc-offer-mining-completed-command.js

```diff
@@ -77,29 +77,12 @@ class DcOfferMiningCompletedCommand extends Command {
                 };
             }
 
-            if (this.config.parentIdentity) {
-                const hasPermission = await this.profileService.hasParentPermission();
-                if (!hasPermission) {
-                    const message = 'Identity does not have permission to use parent identity funds. To replicate data please acquire permissions or remove parent identity from config';
-                    this.logger.warn(message);
-                    throw new Error(message);
-                }
-
-                const hasFunds = await
-                this.dcService.parentHasProfileBalanceForOffer(offer.token_amount_per_holder);
-                if (!hasFunds) {
-                    const message = 'Parent profile does not have enough tokens. To replicate data please deposit more tokens to your profile';
-                    this.logger.warn(message);
-                    throw new Error(message);
-                }
-            } else {
-                const hasFunds =
-                    await this.dcService.hasProfileBalanceForOffer(offer.token_amount_per_holder);
-                if (!hasFunds) {
-                    const message = 'Not enough tokens. To replicate data please deposit more tokens to your profile';
-                    this.logger.warn(message);
-                    throw new Error(message);
-                }
+            const hasFunds =
+                await this.dcService.hasProfileBalanceForOffer(offer.token_amount_per_holder);
```

Comment:
Just opening topic for discussion. It would be better that we move all profile related methods to profile service. For example we shouldn't call dcService.hasProfileBalanceForOffer, this method should be part of profile service. What do you guys think?

Reply:
I think that just going with "if it calls the profile, it's profile service" wouldn't make it any clearer since our services would overlap each other. For example, here DC service / Pricing service / Profile service all make sense to use, since it's a dc, checking if they have funds, on their profile.